### DISCUSSION TITLE
Problem when using optional for enums

### DISCUSSION THREAD

**Discussion by nieg675** [elixir-protobuf/protobuf#300] [Q&A] [answered]
Hello! We are running from commit 48c39d4fd5eb80fcb343d2cdf2dce6d1433e716d to get optional support. However, we experience a problem when using the optional field for enums. If we encode a message that has the optional enum field set to the enum member with tag = 0, we get nil for this field when decoding it. Is this somehow expected, or is it a potential bug?

**Comment by whatyouhide** [comment, marked as answer]
This needs more investigation, and might be a potential bug. Opening an issue for this.

### Edit

My bad, this was actually already fixed in https://github.com/elixir-protobuf/protobuf/pull/302.
